### DISCUSSION TITLE
chainId of length 3 characters does not work

### DISCUSSION THREAD

**Discussion by trantoan960** [CoolBitX-Technology/coolwallet-js-sdk#246] [Q&A] [answered]
# chainId of length 3 characters does not work

- **Description**
we are suspecting the case where chainId is 3 characters length, it will be compile error.
When we test with 2 character chainId everything works back to normal

- **Expected Behavior**
Check the logic of handling the chainId and pass this error.

- **Screenshots or appendix**
![](https://user-images.githubusercontent.com/31679583/149267594-bdab6ddd-dbd8-4f1f-8846-bff1833abfa5.png)

![](https://user-images.githubusercontent.com/31679583/149267642-25471a3e-5acd-416c-b413-0e8392db9159.png)

![](https://user-images.githubusercontent.com/31679583/149267642-25471a3e-5acd-416c-b413-0e8392db9159.png)

**Comment by skchang0710** [comment, marked as answer]
This error does not directly relate to chainId.
The getKeyRecoveryParam method is important to calculate the v value for a signed transaction need. It fails usually because the rawData should be the same as parameters while signing signature and recovering the v value. 
We know it is difficult to observe rawData produced inside the CoolWallet Card, so apdu.tx.getSignedHex method is provided for developers to check the rawData. We recommend to use it in the development phase or data transmission might be oversized.
The apdu.tx.getSignedHex is used in the coin-template sample code. Please give it a read. If you are still not get it work, welcome to paste your code here and we will see if there is anything that we can support.
